File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    float index;
    float L = 0;
    float S = 0;
    int i;
    int sentences = 0;
    float words = 1.0;
    int letters = 0;
    string text = get_string("Text: ");
    int length = strlen(text);

    for (i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }

        else if (text[i] == ' ')
        {
            words++;
        }

        else if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
    }

    L = (letters / words) * 100;
    S = (sentences / words) * 100;

    index = 0.0588 * L - 0.296 * S - 15.8;
    int j = round(index);
    if (j < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (j > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", j);
    }


}

File: sentimental-readability/readability.py
# TODO
import re


def main():
    text = input("Text: ")
    grade = formula(text)
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def formula(text):
    words = len(text.split(" "))
    sentences = len(re.split(r"[.!?]", text)) - 1
    letters = 0
    for i in text:
        if i.isalpha():
            letters += 1

    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


main()
